pkgs/development/beam-modules/default.nix

```diff
@@ -37,9 +37,20 @@ let
         buildMix = callPackage ./build-mix.nix {};
 
         # BEAM-based languages.
-        elixir = if versionAtLeast (lib.getVersion erlang) "18"
-                 then callPackage ../interpreters/elixir { debugInfo = true; }
-                 else throw "Elixir requires at least Erlang/OTP R18.";
+        elixir = defaultScope.elixir-1_4;
```

Comment:
You can drop `default scope.` from all the new code if you add `rec` just before opening "{" of this block. This will also allow you to write `inherit rebar erlang;` instead of `erlang = ...;` and `rebar = ...;` inside args of `lib.callElixir`.

Another option is to replace `defaultScope.` with `self.`.

These have slightly different semantics, but in this case they _seem to be_ equivalent.